### ISSUE TITLE
[BUG]: Issue with the vocoder calling during inference

### ISSUE THREAD

**Issue by ArkhamImp** [open-mmlab/Amphion#105]
## Describe the bug
In [models/base/new_inference.py](https://github.com/open-mmlab/Amphion/blob/57ec89cecb0fa6097d13471a03a6545145221265/models/base/new_inference.py#L145C5-L145C5)

`vocoder_cfg, vocoder_ckpt = self._parse_vocoder(self.args.vocoder_dir)`

If the vocoder checkpoint is saved as '\*.pt' it works fine, but there will be error when I using Hifigan checkpoint which is save as '\*.bin'.

**Comment by VocodexElysium** [comment]
Could you tell me which task you tried to run and encountered this BUG?

Currently, regarding loading neural vocoder for other tasks, both a *.pt checkpoint or a statedict folder that contains a series of *.bin files are supported as in [models/vocoders/vocoder_inference.py](https://github.com/open-mmlab/Amphion/blob/57ec89cecb0fa6097d13471a03a6545145221265/models/vocoders/vocoder_inference.py), the function `load_nn_vocoder`. The function `parse_vocoder` should have been discarded.

**Comment by ArkhamImp** [comment]
So I should not use the `inference` function of In [models/base/new_inference.py](https://github.com/open-mmlab/Amphion/blob/57ec89cecb0fa6097d13471a03a6545145221265/models/base/new_inference.py#L145C5-L145C5) and write my own?

**Comment by VocodexElysium** [comment]
> So I should not use the `inference` function in [models/base/new_inference.py](https://github.com/open-mmlab/Amphion/blob/57ec89cecb0fa6097d13471a03a6545145221265/models/base/new_inference.py#L145C5-L145C5) and write my own?
> 
> Or maybe copy the load model function from [models/vocoders/vocoder_inference.py](https://github.com/open-mmlab/Amphion/blob/a840088a9cc1d5c3afa3ed2f6c39db35c32d1f65/models/vocoders/vocoder_inference.py#L270) is more straightforward.

Yes, you can just write your own inference function with just base names aligned with the new_inference.py

Thanks for your reporting, I will check the compatibility of loading both old .pt vocoder file and new vocoder state dict folder of each downstream tasks this week.

**Comment by mysxs** [comment]
> So I should not use the `inference` function in [models/base/new_inference.py](https://github.com/open-mmlab/Amphion/blob/57ec89cecb0fa6097d13471a03a6545145221265/models/base/new_inference.py#L145C5-L145C5) and write my own?
> 
> Or maybe copy the load model function from [models/vocoders/vocoder_inference.py](https://github.com/open-mmlab/Amphion/blob/a840088a9cc1d5c3afa3ed2f6c39db35c32d1f65/models/vocoders/vocoder_inference.py#L270) is more straightforward.

Hi @VocodexElysium~, excuse me! I would like to ask which .bin file do you use with higfigan? Or how did you change your code?

**Comment by VocodexElysium** [comment]
> > So I should not use the `inference` function in [models/base/new_inference.py](https://github.com/open-mmlab/Amphion/blob/57ec89cecb0fa6097d13471a03a6545145221265/models/base/new_inference.py#L145C5-L145C5) and write my own?
> > Or maybe copy the load model function from [models/vocoders/vocoder_inference.py](https://github.com/open-mmlab/Amphion/blob/a840088a9cc1d5c3afa3ed2f6c39db35c32d1f65/models/vocoders/vocoder_inference.py#L270) is more straightforward.
> 
> Hi @VocodexElysium~, excuse me! I would like to ask which .bin file do you use with higfigan? Or how did you change your code?

Hi @mysxs.

We use the whole folder as a statedict for accelerate loading, so all the .bin files should be used. 

I have checked out the issue#142. You encountered a misuse of a deprecated file. For your case, update your related code snippet with the following can solve your problem:

```python
def _parse_vocoder(vocoder_dir):
        r"""Parse vocoder config"""
        vocoder_dir = os.path.abspath(vocoder_dir)
        ckpt_list = [ckpt for ckpt in Path(vocoder_dir).glob("*.pt")]
        if len(ckpt_list) == 0:
            ckpt_path = vocoder_dir
        else:
            ckpt_list.sort(key=lambda x: int(x.stem), reverse=True)
            ckpt_path = str(ckpt_list[0])
        vocoder_cfg = load_config(
            os.path.join(vocoder_dir, "args.json"), lowercase=True
        )
        return vocoder_cfg, ckpt_path
```

Thanks for your report! We will fix this issue ASAP.
